app/scripts/modules/core/insight/insightFilter.component.ts

```diff
@@ -0,0 +1,16 @@
+import {InsightFilterStateModel} from './insightFilterState.model';
+import {INSIGHT_NGMODULE} from './insight.module';
+
+export class InsightFilterCtrl {
+  constructor(public InsightFilterStateModel: InsightFilterStateModel) {
+  }
+}
+
+INSIGHT_NGMODULE.component('insightFilter', {
```

Comment:
Note the new proposed approach, which imports the NG module and adds the rest of the package's "things" (services, components) to that same ng module.   This would reduce ng module declaration boilerplate in each file.